src/iologindata.cpp

```diff
@@ -536,6 +536,15 @@ bool IOLoginData::loadPlayer(Player* player, DBResult_ptr result)
 		} while (result->next());
 	}
 
+	// load account storage map
+	if (g_game.getNumberOfPlayersByAccount(player->getAccount()) == 0) {
```

Comment:
wouldn't find by account id be enough here? Why checking the players count at all?

Reply:
Indeed, it works when you load the account storage map for the first time. You call the `g_game.getPlayerByAccount(player->getAccount())` and you get a `nullptr` value. However, to be able to save the account storages only once, you need to know how many players are online from the very same account. In this case, the `g_game.getPlayerByAccount(player->getAccount())` call won't help you at all. You will just receive a reference to a player (which, in fact, can be the already logged out player). But, are there more online?

Reply:
Also, `g_game.getPlayerByAccount(player->getAccount())` is an O(n) function too [1]. So, I don't see any big improvement over using the new function either.

===
[1] Worst case scenario.

Reply:
>  [1] Worst case scenario.

It is actually always linear, but the problem persists. Is there nothing on a player object that tags it is online or not?